From fefd0122ca27e5f5b6f8201cfd8eb16e434d909c Mon Sep 17 00:00:00 2001
From: nadlabak <pavel@doshaska.net>
Date: Mon, 17 Apr 2017 02:52:04 +0200
Subject: [PATCH 2/2] GsmCdmaPhone: Fix GSM SIM card ICCID on NV sub CDMA
 devices

The commit 94ae0fe784f0701d3c3bb3c7575362dc6684300b
("GsmCdmaPhone: Return dummy ICCID serial for NV sub")
broke the reported ICCID for GSM SIM cards when used
in global capable NV subscription CDMA devices.

GSM SIM ICCID shouldn't depend on the selected CDMA
subscription source in any way.

Change-Id: Ief48eba854f7c954eaa5132742fe00dc8aa9be13
(cherry picked from commit 71c8c9583cd58434557856b3a657ab531f491a1d)
---
 src/java/com/android/internal/telephony/GsmCdmaPhone.java | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/java/com/android/internal/telephony/GsmCdmaPhone.java b/src/java/com/android/internal/telephony/GsmCdmaPhone.java
index 5abcefb80..b4b55f5b1 100644
--- a/src/java/com/android/internal/telephony/GsmCdmaPhone.java
+++ b/src/java/com/android/internal/telephony/GsmCdmaPhone.java
@@ -734,6 +734,9 @@ public class GsmCdmaPhone extends Phone {
         }
         IccRecords r = mIccRecords.get();
         if (!isPhoneTypeGsm() && r == null) {
+            if (mCdmaSubscriptionSource == CDMA_SUBSCRIPTION_NV) {
+                return DUMMY_NV_ICC_SERIAL;
+            }
             // to get ICCID form SIMRecords because it is on MF.
             r = mUiccController.getIccRecords(mPhoneId, UiccController.APP_FAM_3GPP);
         }
@@ -747,6 +750,9 @@ public class GsmCdmaPhone extends Phone {
         }
         IccRecords r = mIccRecords.get();
         if (!isPhoneTypeGsm() && r == null) {
+            if (mCdmaSubscriptionSource == CDMA_SUBSCRIPTION_NV) {
+                return DUMMY_NV_ICC_SERIAL;
+            }
             // to get ICCID form SIMRecords because it is on MF.
             r = mUiccController.getIccRecords(mPhoneId, UiccController.APP_FAM_3GPP);
         }
-- 
2.17.0

